homeassistant/components/homekit_controller/number.py

```diff
@@ -88,3 +125,70 @@ async def async_set_value(self, value: float):
                 self._char.type: value,
             }
         )
+
+
+class HomeKitEcobeeNumber(CharacteristicEntity, NumberEntity):
+    """Representation of a Number control on a homekit accessory."""
+
+    def __init__(
+        self,
+        conn,
+        info,
+        char,
+        description: HomeKitEcobeeNumberEntityDescription,
+    ):
+        """Initialise a HomeKit number control."""
+        self.entity_description = description
+        super().__init__(conn, info, char)
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return [self._char.type]
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        if name := super().name:
+            return f"{name} - {self.entity_description.name}"
+        return f"{self.entity_description.name}"
+
+    @property
+    def min_value(self) -> float:
+        """Return the minimum value."""
+        return self._char.minValue
+
+    @property
+    def max_value(self) -> float:
+        """Return the maximum value."""
+        return self._char.maxValue
+
+    @property
+    def step(self) -> float:
+        """Return the increment/decrement step."""
+        return self._char.minStep
+
+    @property
+    def value(self) -> float:
+        """Return the current characteristic value."""
+        return self._char.value
+
+    async def async_set_value(self, value: float):
+        """Set the characteristic to this value."""
+        if self.entity_description.first_send_different_value:
```

Comment:
Same as with the button entity. I don't want to parameterize quirks too soon. Ecobee is the only device with these bugs. Make `ECOBEE_SET_HOLD_SCHEDULE` a normal number entity (as it doesn't seem to need quirks). Turn the remaining class into a class just for the fan speed.